scribus/text/storytext.cpp

```diff
@@ -1514,75 +1508,76 @@ int StoryText::firstWord()
 	return pos;
 }
 
-int StoryText::nextWord(int pos)
+int StoryText::nextWord(int pos) const
 {
-	int len = length();
-	//move to text
-	pos = qMin(len, pos+1);
-	//while not at the end, and while we don't find a word boundary, move to the next character
-	while (pos < len  && wordBoundaries.indexOf(text(pos)) < 0)
-		++pos;
-	//while not at the end, and while we find a word boundary, move to the next character
-	while (pos < len  && wordBoundaries.indexOf(text(pos)) >= 0)
-		++pos;
-	//if we didn't get to the end, return current position, otherwise return the end.
-	if (pos < len)
+	BreakIterator* it = getWordIterator();
+	if (!it)
 		return pos;
-	else
-		return len;
+
+	it->setText(plainText().utf16());
+	pos = it->following(pos);
+	pos = it->next();
+	return pos;
 }
 
 int StoryText::prevWord(int pos)
 {
-	pos = qMax(0, pos-1);
-	while (pos > 0 && wordBoundaries.indexOf(text(pos)) < 0)
-		--pos;
-	return wordBoundaries.indexOf(text(pos)) < 0 ? pos + 1 : pos;
+	BreakIterator* it = getWordIterator();
+	if (!it)
+		return pos;
+
+	it->setText(plainText().utf16());
+	pos = it->following(pos);
+	return pos;
 }
 
 int StoryText::endOfWord(int pos) const
 {
-	int len = length();
-	while (pos < len)
-	{
-		if (text(pos).isLetter())
-			++pos;
-		else
-			break;
-	}
+	BreakIterator* it = getWordIterator();
+	if (!it)
+		return pos;
+
+	it->setText(plainText().utf16());
+	pos = it->following(pos);
 	return pos;
 }
 
 int StoryText::endOfSentence(int pos) const
 {
-	int len = length();
-	pos = qMin(len, pos+1);
-	//while not on a sentence boundary, keep moving forward
-	while (pos < len && sentenceBoundaries.indexOf(text(pos)) < 0)
-		++pos;
-	//return the sentence boundary too
-	return pos < len ? pos + 1 : pos;
+	BreakIterator* it = getSentenceIterator();
+	if (!it)
+		return pos;
+
+	it->setText(plainText().utf16());
+	int end = it->following(pos);
+	return end;
 }
 
-int StoryText::nextSentence(int pos)
+int StoryText::nextSentence(int pos) const
 {
-	int len = length();
-	pos = endOfSentence(pos);
-	//while on a sentence boundary, keep moving forward
-	while (pos < len && sentenceBoundaries.indexOf(text(pos)) >= 0)
-		++pos;
-	return pos;
+	BreakIterator* it = getSentenceIterator();
+	if (!it)
+		return pos;
+
+	it->setText(plainText().utf16());
+	pos = it->following(pos);
+	int next = it->next();
```

Comment:
Just make it `pos = it->next()` and return `pos`, no need for a new variable. Similar to code above.